browser-client.cpp

```diff
@@ -381,17 +418,21 @@ void BrowserClient::OnAcceleratedPaint(CefRefPtr<CefBrowser>, PaintElementType t
 	//if (bs->texture)
 	//	gs_texture_acquire_sync(bs->texture, 1, INFINITE);
 
-#else
+#elif defined(__WIN32)
 	bs->texture = gs_texture_open_shared((uint32_t)(uintptr_t)shared_handle);
+#else
+	bs->texture = gs_texture_create_from_dmabuf(bs->width, bs->height, format.drm_format, format.gs_format,
+						    info.plane_count, fds, strides, offsets,
+						    modifier != DRM_FORMAT_MOD_INVALID ? modifiers : NULL);
 #endif
 	UpdateExtraTexture();
 	obs_leave_graphics();
 
 #if defined(__APPLE__) && CHROME_VERSION_BUILD >= 6367
 	bs->last_handle = info.shared_texture_io_surface;
-#elif CHROME_VERSION_BUILD >= 6367
+#elif defined(_WIN32) && CHROME_VERSION_BUILD >= 6367
 	bs->last_handle = info.shared_texture_handle;
-#else
+#elif defined(__APPLE__) && defined(_WIN32)
```

Comment:
How would both `__APPLE__` and `_WIN32` be defined? Was this meant to be inverted?
```cpp
#elif !defined(__APPLE__) && !defined(_WIN32)
```

Reply:
Old shared_texture code path only meant for Windows and macOS, so `&&` should have been `||`. Fixed.